### DISCUSSION TITLE
Getting request headers?

### DISCUSSION THREAD

**Discussion by eden-omb** [withastro/roadmap#50] [Proposal]
I'm pretty sure this is the correct forum for this, do let me know if it's not.

I'd like to read request headers from the Astro server. In particular, `Accept-Language` would be really useful for i18n. I'm aware of the new `navigator.languages`, but it's missing the quality values. This isn't too bad, but I'm sure there's many other useful applications for this. Possibly this could be a good way to handle authentication without setting up an additional server? Let me know what you think.

**Comment by florian-lefebvre** [comment]
Headers can be retrieved from the context: `Astro.headers` or `context.headers`
